x-pack/plugins/maps/public/components/metrics_editor/metric_editor.tsx

```diff
@@ -57,30 +57,39 @@ export function MetricEditor({
     if (!metricAggregationType) {
       return;
     }
-    const newMetricProps = {
```

Comment:
I still think you want to use the spread operator to ensure all keys are passed to onChange. For example, when you add `percentage` key, you do not want specific logic to handle that.